src/components/Products.jsx

```diff
@@ -0,0 +1,35 @@
+import { useEffect, useState } from "react";
+
+export const Products = () => {
+  const [post, setPost] = useState([]);
+  const [location, setLocation] = useState({ latitude: null, longitude: null });
+  useEffect(() => {
+    if (post.length < 1) {
```

Comment:
I would advise setting post initial data as null
and then just check if not null
or just fix check to post.length

Reply:
and wy do you need this check